### PR TITLE
feat: return event on deletion

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

---

- Fixes not deleting events due to data existing in other related tables.
- Batches deletion in transaction.
- Returns event (id) on event deletion - this will be useful in case of updating cache directly after deletion.

### CODE REVIEW

File: server/src/controllers/Events/resolver.ts [hunk 1]
```diff
@@ -402,12 +402,15 @@ ${unsubscribe}
...
-    await prisma.events.delete({ where: { id } });
-    return true;
+  @Mutation(() => Event)
+  async deleteEvent(@Arg('id', () => Int) id: number): Promise<Event> {
+    const result = await prisma.$transaction([
+      prisma.event_sponsors.deleteMany({ where: { event_id: id } }),
+      prisma.rsvps.deleteMany({ where: { event_id: id } }),
+      prisma.user_event_roles.deleteMany({ where: { event_id: id } }),
+      prisma.events.delete({ where: { id } }),
+    ]);
```
Comment (ojeytonwilliams) [inline server/src/controllers/Events/resolver.ts:412]: How about doing this via the db schema?  We can attach the `@relation(...onDelete: Cascade)` attribute to the event_sponsors, rsvps and user_event_roles models and let prisma handle it.

I think it should either all be handled in resolver logic or all handled at the DB level.  We could go either way, but data integrity seems more the database's responsibility than the server's.

Comment (gikf) [reply server/src/controllers/Events/resolver.ts]: Yep, that will make things easier in resolver. I've pushed changes with that.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ojeytonwilliams** [comment]
I only just noticed that we have a resolver, but no way to reach that via the UI.  I think event deletion is *probably* MVP.  At the very least, I'll create an issue.

**Comment by gikf** [comment]
@ojeytonwilliams events deletion is in UI in the event details on `dashboard/events/1`

**Comment by ojeytonwilliams** [comment]
:man_facepalming: how did I forget that... I just looked on `dashboard/events/1/edit`, didn't see a delete button and didn't use my brain.

Anyway thanks for saving me from further foolishness.

**Pull request merged**